src/gap.c

```diff
@@ -1302,41 +1294,16 @@ Obj FuncCALL_WITH_CATCH( Obj self, Obj func, Obj args )
       TLS(CurrStat) = currStat;
       TLS(RecursionDepth) = recursionDepth;
     } else {
-      switch (LEN_PLIST(plain_args)) {
-      case 0: result = CALL_0ARGS(func);
-        break;
-      case 1: result = CALL_1ARGS(func, ELM_PLIST(plain_args,1));
-        break;
-      case 2: result = CALL_2ARGS(func, ELM_PLIST(plain_args,1),
-                                  ELM_PLIST(plain_args,2));
-        break;
-      case 3: result = CALL_3ARGS(func, ELM_PLIST(plain_args,1),
-                                  ELM_PLIST(plain_args,2), ELM_PLIST(plain_args,3));
-        break;
-      case 4: result = CALL_4ARGS(func, ELM_PLIST(plain_args,1),
-                                  ELM_PLIST(plain_args,2), ELM_PLIST(plain_args,3),
-                                  ELM_PLIST(plain_args,4));
-        break;
-      case 5: result = CALL_5ARGS(func, ELM_PLIST(plain_args,1),
-                                  ELM_PLIST(plain_args,2), ELM_PLIST(plain_args,3),
-                                  ELM_PLIST(plain_args,4), ELM_PLIST(plain_args,5));
-        break;
-      case 6: result = CALL_6ARGS(func, ELM_PLIST(plain_args,1),
-                                  ELM_PLIST(plain_args,2), ELM_PLIST(plain_args,3),
-                                  ELM_PLIST(plain_args,4), ELM_PLIST(plain_args,5),
-                                  ELM_PLIST(plain_args,6));
-        break;
-      default: result = CALL_XARGS(func, plain_args);
+    result = CallFuncList(func, args);
+    SET_ELM_PLIST(res,1,True);
```

Comment:
The indention here is wrong (at least two more spaces are needed, as this is in an `else` block). This causes several unnecessary diffs later on.

Off-topic: One of these days, I'd love if we could agree on a common code formatting standard for the kernel ;).